### ISSUE TITLE
Fix false positive in rule LogicConditionNeedOptimizationCheck

### ISSUE THREAD

**Issue by Ravynne** [sevntu-checkstyle/sevntu.checkstyle#275]
The code below produces the error message "Condition with ? at line xxx position 36 need optimization. All method calls are advised to move to end of logic expression."

``` java
    private static void fixMe() {
        String x = "test";
        boolean y = x.length() == 3 ? true : false;
    }
```
